File: backend/models/agency.js
const mongoose = require('mongoose');

const agencySchema = new mongoose.Schema({
  owner: {type: String, required: true},
  description: {type: String, required: true},
  image: {type: String, required: true},
  location: {type: String, required: true},
  date_creation: {type: String, required: true}
});

module.exports = mongoose.model('Agency', agencySchema ,'agencys');

File: mobile/lib/model/agency.dart
import 'package:flutter/material.dart';

class Agency {
  final String id;
  final String owner;
  final String description;
  final String image;
  final String location;
  final String datecreation;


  const Agency({
    required this.id,
    required this.description,
    required this.owner,
    required this.image,
    required this.location,
    required this.datecreation
  });


  factory Agency.fromJson(Map<String, dynamic> json) {
    return Agency(
        id: json['id'],
        owner: json['owner'],
        description: json['description'],
        image: json['image'],
        location: json['location'],
      datecreation: json['datecreation'],
    );
  }
}
